examples/nlp/semantic_similarity_with_keras_nlp.py

```diff
@@ -0,0 +1,274 @@
+semantic similarity with HuggingFace Transformers library. In this tutorial we will 
+learn how to use [KerasNLP](https://keras.io/keras_nlp/), an extension of the core Keras API, 
+for the same task. We'll also learn how KerasNLP reduces the boilerplate code and makes models easy to use.
+
+This guide is broken into following parts:
+
+1. *Setup*, task definition, and establishing a baseline.
+2. *Establishing baseline* with BERT.
+3. *Saving and Reloading* the model.
+4. *Performing inference* with the model.
+5  *Enhancing Performance* with RoBERTa
+
+# Setup
+
+To begin, we can import `keras_nlp`, `keras` and `tensorflow`.
+
+"""
+
+"""shell
+pip install -q keras-nlp
+"""
+
+
+import numpy as np
+import pandas as pd
+
+import tensorflow as tf
+from tensorflow import keras
+import keras_nlp
+import tensorflow_datasets as tfds
+
+"""
+We'll Load the SNLI dataset using tensorflow-datasets library.
+There are more than 550k samples in total. To keep this example running quickly, 
+we will use 20% of trainining samples for this example.
+
+## Overview of SNLI Dataset:
+
+Every sample in the dataset contains three components of form (hypothesis, premise, 
+label).
+
+`premise` is the initial caption provided to the author of the pair, `hypothesis` is 
+the hypothesis caption created by the author of the pair, and `label` is assigned by 
+annotators to denote the similarity between the two sentences.
+
+The dataset includes three possible similarity label values: Contradiction, 
+Entailment, and Neutral, which respectively represent completely dissimilar sentences,
+similar meaning sentences, and neutral sentences where no clear similarity or 
+dissimilarity can be established between them.
+
+"""
+
+snli_train = tfds.load("snli", split="train[:20%]")
+snli_val = tfds.load("snli", split="validation")
+snli_test = tfds.load("snli", split="test")
+
+# Define labels for classification task
+labels = ["neutral", "entailment", "contradiction"]
+
+"""
+### Preprocessing
+
+Tensorflow datasets come as prefetch datasets, with strings converted into bytes. To
+make the tutorial easy to grasp, we convert our dataset to dataframe.
+"""
+
+
+def to_dataframe(dataset):
+    dataset = tfds.as_dataframe(dataset)
+    # decoding bytes to strings
+    dataset["hypothesis"] = dataset["hypothesis"].str.decode("utf-8")
+    dataset["premise"] = dataset["premise"].str.decode("utf-8")
+    return dataset
+
+
+"""
+Let's take a look at how our dataset looks like
+"""
+snli_val_df = to_dataframe(snli_val)
+snli_val_df.head()
+
+"""
+This utility function tokenizes our string inputs according to model's requirements.
+We return tuples of data and labels.
+"""
+
+
+def preprocess_dataset(dataset, preprocessor):
+    dataset = to_dataframe(dataset)
+    # some sample have -1 as label entries. We'll simply drop those here
+    dataset.drop(dataset[dataset["label"] == -1].index, inplace=True)
+
+    x = preprocessor((dataset["hypothesis"], dataset["premise"]))
```

Comment:
If you set up your dataset correctly, you should be able to just do
`dataset = dataset.map(preprocessor, num_parallel_calls=tf.data.AUTOTUNE)`

The results `(x, y)` tuple dataset can be passed directly to fit as a single entity.